src/lib.rs

```diff
@@ -56,9 +55,11 @@ impl CallbackThreadPool {
         for _ in 0..thread_num {
             let receiver = cb_receiver.clone();
             thread_pool.push(thread::spawn(move || {
+                // Create a dedicated runtime for the thread
+                let rt = tokio::runtime::Runtime::new().expect("Unable to create runtime");
```

Comment:
I think this is a reasonable place to panic. 👍

The developer using UPClient can't really do anything if unable to create a tokio Runtime

Reply:
Yes, I don't like panic, but it seems like I can do nothing here...